File: Java/src/main/java/org/codecop/dependencies/subclass_and_override/MarketingCampaign.java
package org.codecop.dependencies.subclass_and_override;

import java.time.DayOfWeek;
import java.time.LocalDateTime;

public class MarketingCampaign {

    public boolean isActive() {
        return System.currentTimeMillis() % 2 == 0;
    }

    public boolean isCrazySalesDay() {
        return dayOfWeek().compareTo(DayOfWeek.FRIDAY) == 0;
    }

    private DayOfWeek dayOfWeek() {
        return LocalDateTime.now().getDayOfWeek();
    }
}

File: PHP/src/A/MarketingCampaign.php
<?php

namespace Org\Codecop\Dependencies\A;

class MarketingCampaign {

    public function isActive(): bool {
        return (floor(microtime(true) * 1000)) % 2 == 0;
    }

    public function isCrazySalesDay(): bool {
        return date('N') == 5; // Friday corresponds to day number 5
    }
}
